plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -836,6 +837,61 @@ private void testSqlServerDatetimeOffset(ZoneId sessionZone)
+                .addRoundTrip("DATETIMEOFFSET(1)", "'1400-09-27 00:00:00.1+07:00'", createTimestampWithTimeZoneType(1), "TIMESTAMP '1400-09-27 00:00:00.1+07:00'")
+                .addRoundTrip("DATETIMEOFFSET(2)", "'1400-09-27 00:00:00.12+07:00'", createTimestampWithTimeZoneType(2), "TIMESTAMP '1400-09-27 00:00:00.12+07:00'")
+                .addRoundTrip("DATETIMEOFFSET(3)", "'1400-09-27 00:00:00.123+07:00'", createTimestampWithTimeZoneType(3), "TIMESTAMP '1400-09-27 00:00:00.123+07:00'")
+                .addRoundTrip("DATETIMEOFFSET(4)", "'1400-09-27 00:00:00.1234+07:00'", createTimestampWithTimeZoneType(4), "TIMESTAMP '1400-09-27 00:00:00.1234+07:00'")
+                .addRoundTrip("DATETIMEOFFSET(5)", "'1400-09-27 00:00:00.12345+07:00'", createTimestampWithTimeZoneType(5), "TIMESTAMP '1400-09-27 00:00:00.12345+07:00'")
+                .addRoundTrip("DATETIMEOFFSET(6)", "'1400-09-27 00:00:00.123456+07:00'", createTimestampWithTimeZoneType(6), "TIMESTAMP '1400-09-27 00:00:00.123456+07:00'")
+                .addRoundTrip("DATETIMEOFFSET(7)", "'1400-09-27 00:00:00.1234567+07:00'", createTimestampWithTimeZoneType(7), "TIMESTAMP '1400-09-27 00:00:00.1234567+07:00'")
+                .execute(getQueryRunner(), session, sqlServerCreateAndInsert("test_sqlserver_datetimeoffset_old_date"));
+    }
+
+    @Test
+    public void testSqlServerDatetimeOffsetOldDatesRangeQuery()
+    {
+        // Tests the custom predicate push down controller for DATETIMEOFFSET types with values before and after 1583
+        List<String> dateTimeOffsetValues = List.of(
+                "'1400-01-01 00:00:00.1234567+00:00'",
+                "'1500-01-01 00:00:00.1234567+00:00'",
+                "'1582-12-31 23:59:59.9999999+00:00'",
+                "'1583-01-01 00:00:00+00:00'",
+                "'1583-01-01 00:00:00.1234567+00:00'",
+                "'1600-01-01 00:00:00.1234567+00:00'",
+                "'1700-01-01 00:00:00.1234567+00:00'",
+                "'1800-01-01 00:00:00.1234567+00:00'",
+                "'1900-01-01 00:00:00.1234567+00:00'");
+
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_sqlserver_datetimeoffset_old_date_range_query", "(col0 datetimeoffset(7))", dateTimeOffsetValues)) {
+            assertQuery("SELECT count(*) FROM " + table.getName(), "SELECT 9");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 <= TIMESTAMP '1582-12-31 23:59:59.9999999+00:00'", "SELECT 3");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 >= TIMESTAMP '1583-01-01 00:00:00+00:00'", "SELECT 6");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 IN (TIMESTAMP '1582-12-31 23:59:59.9999999+00:00', TIMESTAMP '1583-01-01 00:00:00+00:00')", "SELECT 2");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 IN (TIMESTAMP '1583-01-01 00:00:00+00:00', TIMESTAMP '1600-01-01 00:00:00.1234567+00:00')", "SELECT 2");
```

Comment:
How about additional test cases for `NOT IN` and `BETWEEN`

Reply:
Added additional test casess